packages/native-ui/src/components/custom/Spinner/Spinner.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect } from 'react';
+import {
+  useSharedValue,
+  withTiming,
+  useAnimatedProps,
+  useAnimatedStyle,
+  Easing,
+  withSequence,
+  withRepeat,
+} from 'react-native-reanimated';
+import { G } from 'react-native-svg';
+import { Platform } from 'react-native';
+import { StyledCircle, StyledSpinner, StyledSvg } from './styled-components';
+
+interface SpinnerProps {
+  size?: 'xs' | 'sm' | 'md' | 'lg';
+  color?: string;
+}
+
+const getWidth = (size: 'xs' | 'sm' | 'md' | 'lg') => {
+  switch (size) {
+    case 'xs':
+      return 14;
+    case 'sm':
+      return 20;
+    case 'md':
+      return 28;
+    case 'lg':
+      return 40;
+    default:
+      return 32;
+  }
+};
+
+const Spinner: React.FC<SpinnerProps> = ({ size = 'md', color }) => {
+  const width = getWidth(size);
+  const CIRCUMFERENCE = (width - 4) * Math.PI;
+  const R = CIRCUMFERENCE / (2 * Math.PI);
+  const STROKE_WIDTH = 2;
```

Comment:
I think this is the one that the size makes it conditional. 

> so it's 2px on > 16px and 1.5px on < 16px

(from slack) this is probably what @rlanguw saw